### PR TITLE
Feature: Add settings struct

### LINKED ISSUE #37: Add struct für `get_settings` XML-RPC API endpoint
### Is your feature request related to a problem?

As a consumer of the Cobbler API  
I want to be able to get the full settings information from Cobbler  
so that I can show it to the user.

### Provide a detailed description of the proposed feature

Expose a custom struct that contains all the settings data.

### Alternatives you've considered

None

### Additional information

None

### PR DESCRIPTION
**Pull request by SchoolGuy**
Fixes #37 

The issue with this PR is that the Cobbler 3.3.x release series changed too much in the settings to be compatible with the Golang stability guarantees. As such this struct will fail to work with some 3.3.x versions of Cobbler. I haven't verified what version breaks the struct but with the currently available test fixture, the method works as expected.

We break another secondary best practice in Golang with this struct: Using `New<Type>` functions to create complex structs. This cannot be done since the mapstructure library needs an empty struct and would - according to my understanding of the docs - anyway discard the values that we pass it since it only takes over the fields for the newly created `interface{}` that a consumer can then later cast to the expected struct type.

### CODE REVIEW

**Comment by SchoolGuy** [comment]
I have no strong opinion but I think it may be worthwhile to discuss the fact that we may be able to group certain settings into sub-structs and use the embedding functionality of the mapstructure library to create the server-side representation of the object for parsing purposes.

File: settings.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	if err := decoder.Decode(raw); err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
```
Comment (tiltingpenguin) [inline settings.go]: Ideally we would deduplicate this with cobblerSignatureHacks in signatures.go, but for now I would be ok with just changing the f and t variable names

**Review by tiltingpenguin — COMMENTED** [review]
I only have one nitpick, otherwise LGTM

**Comment by tiltingpenguin** [comment]
Regarding the comment, as I don't know a lot about the mapstructure library I can't comment to much, but I think if we can group settings and nest them to make parsing easier this is worth looking at in the future. For now I would accept this PR as it is

**Review by tiltingpenguin — APPROVED** [review]
LGTM

**Pull request merged**